src/core/BufferGeometry.js

```diff
@@ -643,7 +643,7 @@ class BufferGeometry extends EventDispatcher {
 
 			let normalAttribute = this.getAttribute( 'normal' );
 
-			if ( normalAttribute === undefined ) {
+			if ( normalAttribute === undefined || positionAttribute.count !== normalAttribute.count ) {
```

Comment:
This change will produce a memory leak since the old normal buffer attribute is not properly disposed of. Once an attribute has been set, it's not correct to overwrite it. It's necessary to call `dispose()` on the geometry. 

Besides, a length check is also not ideal since buffer attributes are not supposed to change in size. They are supposed to be fixed-size.